tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.C

```diff
@@ -0,0 +1,279 @@
+/// \file
+/// \ingroup tutorial_dataframe
+/// \notebook -draw
+/// The Higgs to four lepton analysis from the ATLAS Open Data release of 2020, with RDataFrame.
+///
+/// This tutorial is the Higgs to four lepton analysis from the ATLAS Open Data release in 2020
+/// (http://opendata.atlas.cern/release/2020/documentation/). The data was taken with the ATLAS detector
+/// during 2016 at a center-of-mass energy of 13 TeV. The decay of the Standard Model Higgs boson
+/// to two Z bosons and subsequently to four leptons is called the "golden channel". The selection leads
+/// to a narrow invariant mass peak on top a relatively smooth and small background, revealing
+/// the Higgs at 125 GeV.
+/// The analysis is translated to an RDataFrame workflow processing about 300 MB of simulated events and data.
+///
+/// Lepton scale factors are applied to leptons in MC simulations to correct for the differences in the
+/// reconstruction, identification and trigger mechanisms used for MC samples and real data.
+/// Systematic uncertainties for those MC scale factors are evaluated and the Vary function of RDataFrame is used to
+/// scale the variations to the final four leptons mass distribution.
+///
+///
+/// \macro_code
+/// \macro_image
+/// \date March 2020
+/// \author Stefan Wunsch (KIT, CERN)
+/// \date August 2022
+/// \author Julia Mathe (CERN)
+/// \date August 2023
+/// \author Marta Czurylo (CERN)
+
+#include <Math/Vector4D.h>
+#include "ROOT/RDFHelpers.hxx"
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RVec.hxx"
+#include "TCanvas.h"
+#include "TH1D.h"
+#include "TLatex.h"
+#include <TLegend.h>
+#include "TProfile.h"
+#include "TStyle.h"
+
+using namespace ROOT::VecOps;
+using cRVecF = const ROOT::RVecF &;
+using namespace ROOT::RDF::Experimental;
+
+// Define functions needed in the analysis
+// Select events for the analysis
+bool GoodElectronsAndMuons(const ROOT::RVecI &type, cRVecF pt, cRVecF eta, cRVecF phi, cRVecF e, cRVecF trackd0pv,
+                           cRVecF tracksigd0pv, cRVecF z0)
+{
+   for (size_t i = 0; i < type.size(); i++) {
+      ROOT::Math::PtEtaPhiEVector p(pt[i] / 1000.0, eta[i], phi[i], e[i] / 1000.0);
```

Comment:
```suggestion
      ROOT::Math::PtEtaPhiEVector p(0.001 * pt[i], eta[i], phi[i], 0.001 * e[i]);
```
Float divisions are not so performant, so it's good practice to avoid them. By being pedantic in the tutorials, our students also learn better C++ :)

Applies also to other places in the code